File: python/src/twentyfifteen/day_01.py
from python.src import lib


def day_one(part: int):
    content = lib.read_input(year=2015, day=1, part=part)
    if part == 1:
        return day_one_calculate_part_one(content)

    return day_one_calculate_part_two(content)


def day_one_calculate_part_one(content: str) -> int:
    return content.count("(") - content.count(")")


def day_one_calculate_part_two(content: str) -> int:
    value = 0
    for index in range(0, len(content)):
        character = content[index]
        if character == "(":
            value += 1
        else:
            value -= 1

        if value == -1:
            return index + 1

    return value

File: go/twentyfifteen/day01.go
package twentyfifteen

import "strings"

func day01Part01(input string) int {
	return strings.Count(input, "(") - strings.Count(input, ")")
}

func day01Part02(input string) int {
	value := 0
	for i, character := range input {
		character := string(character)
		if character == "(" {
			value++
		} else {
			value--
		}

		if value == -1 {
			return i + 1
		}
	}
	return value
}
